src/Pelagos/Bundle/AppBundle/Security/LoginFormAuthenticator.php

```diff
@@ -227,16 +267,51 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
      * @param Request                 $request   A Symfony Request, req by interface.
      * @param AuthenticationException $exception The exception thrown.
      *
+     * @throws AuthenticationException When the user is not found, and credentials is invalid.
+     *
      * @return Response The response or null to continue request.
      */
     public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
```

Comment:
If we really wanted use a different exception for password expired, then `AuthenticationException` hint can be changed to `Exception` hint.

Reply:
NM, this won't work, the interface doesn't like that.